internal/iam/query.go

```diff
@@ -309,6 +309,151 @@ const (
+    with
+    users (id) as (
+      select public_id
+        from iam_user
+       where public_id = any(@user_ids)
+    ),
+    user_groups (id) as (
+      select group_id
+        from iam_group_member_user
+       where member_id in (select id
+                             from users)
+    ),
+    user_accounts (id) as (
+      select public_id
+        from auth_account
+       where iam_user_id in (select id
+                               from users)
+    ),
+    user_oidc_managed_groups (id) as (
+      select managed_group_id
+        from auth_oidc_managed_group_member_account
+       where member_id in (select id
+                             from user_accounts)
+    ),
+    user_ldap_managed_groups (id) as (
+      select managed_group_id
+        from auth_ldap_managed_group_member_account
+       where member_id in (select id
+                             from user_accounts)
+    ),
+    managed_group_roles (role_id) as (
+      select distinct role_id
+        from iam_managed_group_role
+       where principal_id in (select id
+                                from user_oidc_managed_groups)
+          or principal_id in (select id
+                                from user_ldap_managed_groups)
+    ),
+    group_roles (role_id) as (
+      select role_id
+        from iam_group_role
+       where principal_id in (select id
+                                from user_groups)
+    ),
+    user_roles (role_id) as (
+      select role_id
+        from iam_user_role
+       where principal_id in (select id
+                                from users)
+    ),
+    user_group_roles (role_id) as (
+      select role_id
+        from group_roles
+       union
+      select role_id
+        from user_roles
+       union
+      select role_id
+        from managed_group_roles
+    ),
+    roles_with_grants (role_id, canonical_grant) as (
+      select iam_role_grant.role_id,
+             iam_role_grant.canonical_grant
+        from iam_role_grant
+        join iam_role
+          on iam_role.public_id = iam_role_grant.role_id
+        join iam_grant
+          on iam_grant.canonical_grant = iam_role_grant.canonical_grant
+       where iam_role.public_id in (select role_id
+                                      from user_group_roles)
+         and iam_grant.resource = any(@resources)
+    ),
+	global_role_special_grants as (
+      select iam_role_global.public_id             as role_id,
+             iam_role_global.scope_id              as role_scope_id,
+             'global'                              as role_parent_scope_id,
+             iam_role_global.grant_scope           as grant_scope,
+             roles_with_grants.canonical_grant     as canonical_grant
+        from iam_role_global
+        join roles_with_grants
+          on roles_with_grants.role_id = iam_role_global.public_id
+       where iam_role_global.grant_scope = any('{ children, descendants }')
+    ),
+    global_role_individual_grants as (
```

Comment:
nitpick: add `_scope` for all CTEs that look for `grant_scope` to avoid confusion with queries looking for `iam_grant`  
```suggestion
    global_role_individual_grant_scopess as (
```